sentinel1_reader/sentinel1_burst_slc.py

```diff
@@ -0,0 +1,190 @@
+                                                 width,
+                                                 ref_epoch)
+
+    def to_vrt_file(self, out_path):
+        '''
+        Write burst to VRT file.
+
+        Parameters:
+        -----------
+        out_path : string
+            Path of output VRT file.
+        '''
+        line_offset = self.i_burst * self.shape[0]
+
+        inwidth = self.last_valid_sample - self.first_valid_sample
+        inlength = self.last_valid_line - self.first_valid_line
+        outlength, outwidth = self.shape
+        yoffset = line_offset + self.first_valid_line
+        localyoffset = self.first_valid_line
+        xoffset = self.first_valid_sample
+        gdal_obj = gdal.Open(self.tiff_path, gdal.GA_ReadOnly)
+        fullwidth = gdal_obj.RasterXSize
+        fulllength = gdal_obj.RasterYSize
+
+        # TODO maybe cleaner to write with ElementTree
+        tmpl = f'''<VRTDataset rasterXSize="{outwidth}" rasterYSize="{outlength}">
+    <VRTRasterBand dataType="CFloat32" band="1">
+        <NoDataValue>0.0</NoDataValue>
+        <SimpleSource>
+            <SourceFilename relativeToVRT="1">{self.tiff_path}</SourceFilename>
+            <SourceBand>1</SourceBand>
+            <SourceProperties RasterXSize="{fullwidth}" RasterYSize="{fulllength}" DataType="CInt16"/>
+            <SrcRect xOff="{xoffset}" yOff="{yoffset}" xSize="{inwidth}" ySize="{inlength}"/>
+            <DstRect xOff="{xoffset}" yOff="{localyoffset}" xSize="{inwidth}" ySize="{inlength}"/>
+        </SimpleSource>
+    </VRTRasterBand>
+</VRTDataset>'''
+
+        with open(out_path, 'w') as fid:
+            fid.write(tmpl)
+
+    def get_sensing_mid(self):
+        '''
+        Returns sensing mid as datetime object.
+
+        Returns:
+        --------
+        _ : datetime
+            Sensing mid as datetime object.
+        '''
+        d_seconds = 0.5 * (self.shape[0] - 1) * self.azimuth_time_interval
+        return self.sensing_start + datetime.timedelta(seconds=d_seconds)
+
+    def get_isce3_orbit(self, orbit_dir: str):
+        '''
+        Init and return ISCE3 orbit.
+
+        Parameters:
+        -----------
+        orbit_dir : string
+            Path to directory containing orbit files.
+
+        Returns:
+        --------
+        _ : datetime
+            Sensing mid as datetime object.
+        '''
+        if not os.path.isdir(orbit_dir):
+            raise NotADirectoryError
+
+        # determine start and end time from metadata
+        pulse_length = (self.shape[0] - 1) * self.azimuth_time_interval
+        t_pulse_end = self.sensing_start + datetime.timedelta(seconds=pulse_length)
+
+        # find files with self.platform_id
+        item_valid = lambda item, sat_id: os.path.isfile(item) and sat_id in item
+        orbit_files = [item for item in os.listdir(orbit_dir)
+                       if item_valid(f'{orbit_dir}/{item}', self.platform_id)]
+        if not orbit_files:
```

Comment:
Good point @vbrancat I was actually thinking the same. But then I was hesitant if we want to have the reader to be isolated as a pure reader and let workflows at higher level be worried about downloading orbits and all the junk around (connection failure etc). I don't have a strong opinion really but I just wanted to make sure you also note that point. 
Anyways I think the reader should accept external orbit and that should be the default behavior.  
A middle ground would be to have a flag for orbit_download which is off by default. So if the orbit is not found and if the user specifically asks to download, then the reader attempts to download.

Reply:
I agree. The higher-level workflow should **properly** handle this (e.g. check the internet connection and all this stuff). However, I do not see any harm in providing an optional functionality to download the orbit (if not present). Maybe the functionality (it can be a simple flag) should not have too much logic in it (e.g. not checking the internet connection). I will let @LiangJYu illuminate us if this solution does not violate the "separation of functionalities" principle (https://en.wikipedia.org/wiki/Single-responsibility_principle)

Reply:
Sounds good to me as far as it is optional to use.

Reply:
Especially when running in production, I also agree the higher-level workflow should download the orbits. For other use cases, I'm not opposed to including a download option. Perhaps something like the following?
```python
def get_local_orbit_file(t_start, t_stop, orbit_dir):
    # try to find and return path file in orbit_dir else return empty string

def get_online_orbit_file(t_start, t_stop):
    # try to find and return file from online source else return empty string

def get_isce3_orbit(..., online_ok=False):
    ...
    # try local first
    orbit_file = get_local_orbit_file(...)

    # try online if nothing local and ok to go online
    if not orbit_file and online_ok:
        orbit_file = get_online_orbit_file(...)

        # check if nothing found online
        if not orbit_file:
            raise ValueError("no orbit file found")
    ...
```

Reply:
This looks good to me.